packages/core/src/components/collapse/collapse.tsx

```diff
@@ -223,10 +223,12 @@ export class Collapse extends AbstractPureComponent2<ICollapseProps, ICollapseSt
             });
             this.setTimeout(() => this.onDelayedStateChange(), transitionDuration);
         } else if (animationState === AnimationStates.CLOSING_START) {
+            const { clientHeight } = this.contents;
             this.setTimeout(() =>
                 this.setState({
                     animationState: AnimationStates.CLOSING,
                     height: "0px",
+                    heightWhenOpen: clientHeight,
```

Comment:
hmm I'm not sure why this state field is not already set at this time by L222, but this change seems to fix the bug so I'll approve it and look into any potential tweaks later.

Reply:
I'll sign the cla in 1-2 days, because my company wants me to sign the corporate version. I'll send it asap